src/lib/rmt/ssl/certificate_generator.rb

```diff
@@ -96,13 +114,22 @@ def generate(common_name, alt_names)
       '-out', @ssl_paths[:server_csr], '-config', @ssl_paths[:server_config]
     )
 
-    RMT::Execute.on_target!(
-      'openssl', 'x509', '-req', '-in', @ssl_paths[:server_csr], '-out', @ssl_paths[:server_certificate],
-      '-CA', @ssl_paths[:ca_certificate], '-CAkey', @ssl_paths[:ca_private_key],
-      '-days', OPENSSL_SERVER_CERT_VALIDITY_DAYS, '-sha256',
-      '-CAcreateserial',
-      '-extensions', 'v3_server_sign', '-extfile', @ssl_paths[:server_config]
-    )
+    if !ca_password.empty?
+      RMT::Execute.on_target!(
+        'openssl', 'x509', '-req', '-in', @ssl_paths[:server_csr], '-out', @ssl_paths[:server_certificate],
+        '-CA', @ssl_paths[:ca_certificate], '-CAkey', @ssl_paths[:ca_private_key],
+        '-passin', 'stdin', '-days', OPENSSL_SERVER_CERT_VALIDITY_DAYS, '-sha256',
+        '-CAcreateserial', '-extensions', 'v3_server_sign', '-extfile', @ssl_paths[:server_config],
+        stdin: ca_password
+      )
+    else
+      RMT::Execute.on_target!(
+        'openssl', 'x509', '-req', '-in', @ssl_paths[:server_csr], '-out', @ssl_paths[:server_certificate],
+        '-CA', @ssl_paths[:ca_certificate], '-CAkey', @ssl_paths[:ca_private_key],
+        '-days', OPENSSL_SERVER_CERT_VALIDITY_DAYS, '-sha256',
+        '-CAcreateserial', '-extensions', 'v3_server_sign', '-extfile', @ssl_paths[:server_config]
+      )
```

Comment:
Seems like this code isn't reachable, since it is not possible to input an empty password in the dialog.
If that's the case, it should be removed (we don't need CAs with empty passwords).

Reply:
for the case when CA private key is without password (e. g. generated earlier) but server key has to be regenerated. there we skip running password dialog but pass empty one. 
see https://github.com/SUSE/yast2-rmt/pull/19/files#diff-7636ecc7dfb1e414d37898003b656293R96

Reply:
Ah, I see.